src/imitation/algorithms/preference_comparisons.py

```diff
@@ -1198,55 +1198,54 @@ def __init__(
         Raises:
             TypeError: if model is not a RewardEnsemble.
         """
-        if not isinstance(model, reward_nets.RewardEnsemble):
+        if not preference_model.is_ensemble:
             raise TypeError(
-                f"RewardEnsemble expected by EnsembleTrainer not {type(model)}.",
+                "Preference Model of a RewardEnsemble expected by EnsembleTrainer.",
             )
 
         super().__init__(
-            model,
+            preference_model,
             loss,
             batch_size,
             epochs,
             lr,
             weight_decay,
             custom_logger,
         )
-        self.rng = np.random.default_rng(seed=seed)
-
-    def _training_inner_loop(
-        self,
-        fragment_pairs: Sequence[TrajectoryPair],
-        preferences: np.ndarray,
-    ) -> th.Tensor:
-        assert len(fragment_pairs) == preferences.shape[0]
-        losses = []
-        metrics = []
-        for member_idx in range(self._model.num_members):
-            # sample fragments for training via bagging
-            sample_idx = self.rng.choice(
-                np.arange(len(fragment_pairs)),
-                size=len(fragment_pairs),
-                replace=True,
+        self.member_trainers = []
+        for member_pref_model in self._preference_model.member_pref_models:
+            reward_trainer = BasicRewardTrainer(
+                member_pref_model,
+                loss,
+                batch_size,
+                epochs,
+                lr,
+                weight_decay,
+                self.logger,
             )
-            sample_fragments = [fragment_pairs[i] for i in sample_idx]
-            sample_preferences = preferences[sample_idx]
-            output = self.loss.forward(sample_fragments, sample_preferences, member_idx)
-            losses.append(output.loss)
-            metrics.append(output.metrics)
-        losses = th.stack(losses)
-        loss = losses.sum()
+            self.member_trainers.append(reward_trainer)
+        self.rng = th.Generator()
+        if seed:
+            self.rng = self.rng.manual_seed(seed)
 
-        self.logger.record("loss", loss.item())
-        self.logger.record("loss_std", losses.std().item())
+    @RewardTrainer.logger.setter
+    def logger(self, custom_logger):
+        self._logger = custom_logger
+        for member_trainer in self.member_trainers:
+            member_trainer._logger = custom_logger
```

Comment:
We have a setter in `RewardTrainer` so I think no need to access the private variable.
```suggestion
            member_trainer.logger = custom_logger
```